src/signing.rs

```diff
@@ -32,28 +34,15 @@ pub trait AgentSigner {
 
 /// Signs an unsigned zome call using the provided signing implementation
 pub(crate) async fn sign_zome_call(
-    zome_call_unsigned: ZomeCallUnsigned,
+    params: ZomeCallParams,
     signer: Arc<dyn AgentSigner + Send + Sync>,
-) -> Result<ZomeCall> {
-    let pub_key = zome_call_unsigned.provenance.clone();
-
-    let data_to_sign = zome_call_unsigned.data_to_sign().map_err(|e| {
-        anyhow::anyhow!("Failed to get data to sign from unsigned zome call: {}", e)
-    })?;
-
+) -> Result<ZomeCallParamsSigned> {
+    let pub_key = params.provenance.clone();
+    let (_, bytes_hash) = params.serialize_and_hash()?;
     let signature = signer
-        .sign(&zome_call_unsigned.cell_id, pub_key, data_to_sign)
+        .sign(&params.cell_id, pub_key, bytes_hash.into())
         .await?;
 
-    Ok(ZomeCall {
-        cell_id: zome_call_unsigned.cell_id,
-        zome_name: zome_call_unsigned.zome_name,
-        fn_name: zome_call_unsigned.fn_name,
-        payload: zome_call_unsigned.payload,
-        cap_secret: zome_call_unsigned.cap_secret,
-        provenance: zome_call_unsigned.provenance,
-        nonce: zome_call_unsigned.nonce,
-        expires_at: zome_call_unsigned.expires_at,
-        signature,
-    })
+    let bytes = ExternIO::encode(params)?;
```

Comment:
This is a little risky, serializing again after hashing+signing. Does `serialize_and_hash` not return this value and it's just currently an `_` above?

Reply:
Yeah, I've changed it. Thanks because I actually tried it a bunch of times and couldn't get it to work and then I just tried it again based on your comment and it worked first time :sweat_smile: I think I was doing `ExternIO::encode(bytes)` before instead of just `ExternIO(bytes)` which does work.